runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java

```diff
@@ -175,24 +178,52 @@ public void load()
 		{
 			log.warn("Unable to update configuration on disk", ex);
 		}
+
+		if (configWatcherTask != null)
+		{
+			configWatcherTask.cancel(true);
+		}
+
+		configWatcherTask = executor.scheduleAtFixedRate(new ConfigWatcher(propertiesFile, () ->
+		{
+			final Properties properties = new Properties();
+			loadProperties(properties);
+
+			Map<String, String> copy = (Map) ImmutableMap.copyOf(properties);
+			copy.forEach((groupAndKey, value) ->
```

Comment:
This needs to be a map difference or something, this won't work at all for removed config items.

Reply:
Done (this.properties.keySet().retainAll(copy.keySet());)